packages/taiko-client/driver/driver.go

```diff
@@ -385,87 +399,152 @@ func (d *Driver) cacheLookaheadLoop() {
 			d.PreconfHandoverSkipSlots,
 			d.rpc.L1Beacon.SlotsPerEpoch,
 		)
+		wasSequencer bool = false
 	)
 
-	for {
-		select {
-		case <-d.ctx.Done():
+	checkHandover := func(epoch, slot uint64, wasSequencer *bool) {
+		if d.p2pNode == nil {
 			return
-		case <-ticker.C:
-			var (
-				currentEpoch     = d.rpc.L1Beacon.CurrentEpoch()
-				currentSlot      = d.rpc.L1Beacon.CurrentSlot()
-				slotInEpoch      = d.rpc.L1Beacon.SlotInEpoch()
-				slotsLeftInEpoch = d.rpc.L1Beacon.SlotsPerEpoch - d.rpc.L1Beacon.SlotInEpoch()
-			)
+		}
 
-			latestSeenBlockNumber, err := d.rpc.L1.BlockNumber(d.ctx)
-			if err != nil {
-				log.Error("Failed to fetch the latest L1 head for lookahead", "error", err)
-				continue
-			}
+		err := d.preconfBlockServer.CheckLookaheadHandover(d.PreconfOperatorAddress, slot)
+		isSequencer := err == nil
+
+		if isSequencer && !*wasSequencer {
```

Comment:
does `!*wasSequencer` means `wasSequencer` will never be `nil`? otherwise it will panic?

Reply:
because the arg is passed into the function but we can actually ignore that and treat it the same as `lastSeenBlockNumber` here and not pass it into the closure